chain/lf3/f3.go

```diff
@@ -133,43 +136,44 @@ func (fff *F3) runSigningLoop(ctx context.Context) {
 		return nil
 	}
 
-	leaseMngr := new(leaseManager)
 	msgCh := fff.inner.MessagesToSign()
 
 loop:
 	for ctx.Err() == nil {
 		select {
 		case <-ctx.Done():
 			return
-		case l := <-fff.newLeases:
-			// resultCh has only one user and is buffered
-			l.resultCh <- leaseMngr.UpsertDefensive(l.minerID, l.newExpiration, l.oldExpiration)
-			close(l.resultCh)
 		case mb, ok := <-msgCh:
 			if !ok {
 				continue loop
 			}
-
-			for _, minerID := range leaseMngr.Active() {
-				err := participateOnce(ctx, mb, minerID)
-				if err != nil {
-					log.Errorf("while participating for miner f0%d: %+v", minerID, err)
+			participants := fff.leaser.getParticipantsByInstance(mb.Payload.Instance)
+			for _, id := range participants {
+				if err := participateOnce(ctx, mb, id); err != nil {
+					log.Errorf("while participating for miner f0%d: %+v", id, err)
 				}
 			}
 		}
 	}
 }
 
+func (fff *F3) GetOrRenewParticipationToken(ctx context.Context, minerID uint64, previous api.F3ParticipationToken) (api.F3ParticipationToken, error) {
+	select {
+	case <-ctx.Done():
+		return api.F3ParticipationToken{}, ctx.Err()
```

Comment:
Any reason to check the context?

Reply:
To respect the context that is passed through even if successive calls don't take a context? I think that's better than ignoring the context entirely. But I have no strong opinions.

Reply:
We're not blocking, right? IMO, checking the context makes sense if you have a long-running loop, block, etc. Otherwise, it's up to the caller to deal with it.

Reply:
On the subjective matters: It is my opinion that if context exist it should be respected. e.g. if an expired context is passed to a function, regardless of whether it is blocking or not, it should suspend execution.

But I will revert the changes here and ignore the context since it seems this repo does not follow that practice.